examples/example_fec_caching.py

```diff
@@ -0,0 +1,99 @@
+"""
+Example demonstrating the use of the caching decorator with functional equivalance checking
+===========================================================================================
+
+Caches the results of fitness evaluations in a pickle file
+('example_fec_caching_cache.pkl'). To illustrate its practical use,
+compare the runtime of this script when you first call it vs. the
+second time and when you comment out the decorator on
+`inner_objective`.
+
+"""
+
+import time
+
+import numpy as np
+
+import cgp
+
+# %%
+# We define the target function for this example.
+
+
+def f_target(x):
+    return x ** 2 + x + 1.0
+
+
+# %%
+# We then define the objective function for the evolutionary
+# algorithm: It consists of an inner objective which we wrap with the
+# caching decorator. This decorator specifies a pickle file that will be used for
+# caching results of fitness evaluations. The inner objective is then used by the objective
+# function to compute (or retrieve from cache) the fitness of the individual.
+
+
+@cgp.utils.disk_cache("example_fec_caching_cache.pkl", use_fec=True, fec_seed=12345)
```

Comment:
I think it would be instructive to use the other keyword arguments of fec caching here, too.